.github/workflows/release.yml

```diff
@@ -0,0 +1,146 @@
+name: Release
+
+on:
+  workflow_dispatch:
+
+permissions:
+  contents: write
+
+jobs:
+  verify-branch:
+    name: Verify that runs on the main branch
```

Comment:
Note: It looks that running `taiki-e/create-gh-release` always creates a release tag from the default branch, as it uses `gh release create`, [here](https://github.com/taiki-e/create-gh-release-action/blob/72d65cee1f8033ef0c8b5d79eaf0c45c7c578ce3/main.sh#L168) is the exact line. The only exception occurs if the tag for the intended release already exists. The point is that I decided to add this check to prevent a situation where running this workflow on a branch other than main would build assets from this branch but create a release tag from the latest commit on the main branch. While this introduces some limitations, I think it's better to have this explicitly highlighted here.